reactor-core/src/main/java/reactor/core/publisher/FluxPublish.java

```diff
@@ -120,15 +127,24 @@ public void subscribe(CoreSubscriber<? super T> actual) {
 				c = u;
 			}
 
+			inner.parent = c;
+
 			if (c.add(inner)) {
 				if (inner.isCancelled()) {
 					c.remove(inner);
 				}
-				else {
-					inner.parent = c;
-				}
 				c.drain();
 				break;
+			} else if (!this.resetUponSourceTermination) {
+				if (c.error != null) {
+					inner.actual.onError(c.error);
+				} else {
+					if (c.queue.isEmpty()) {
+						c.drain();
```

Comment:
I'm wondering about that one, from a quick glance. is the "don't drain if not empty" on purpose ?

Reply:
That is in fact a mistake. What's more I think it's redundant:
1. `add` will return `false` when the `subscribers == TERMINATED`;
2. the above happens only when the queue is empty

I think I can remove this (currently incorrect of course) defensive drain.